src/mbgl/gl/object_store.hpp

```diff
@@ -75,8 +66,13 @@ class ObjectStore : private util::noncopyable {
     }
 
     UniqueTexture createTexture() {
-        GLuint id = 0;
-        MBGL_CHECK_ERROR(glGenTextures(1, &id));
+        if (pooledTextures.empty()) {
+            pooledTextures.resize(TextureMax);
+            MBGL_CHECK_ERROR(glGenTextures(TextureMax, pooledTextures.data()));
+        }
+
+        GLuint id = pooledTextures.back();
```

Comment:
If there are `TextureMax` textures in use, then `pooledTextures` will be `empty()` and the next `createTexture()` request will create `TextureMax` more textures.